crypto/cms/cms_sd.c

```diff
@@ -706,11 +706,23 @@ int CMS_SignerInfo_sign(CMS_SignerInfo *si)
         si->pctx = pctx;
     }
 
+    /*
```

Comment:
That appears to be true in the source code..
I have thought the same thing about some other controls only to find they were used by external engines? Could that be possible?

Reply:
Yes, its possible although I couldn't find any evidence of that. There was a case of this with the MICALG ctrl which was not used internally, but *was* used by the GOST engine (remembering of course that the GOST engine *used* to be part of our source). But I specifically checked GOST, and its not used there either. I think it is quite unlikely to be used by engines implementing algorithms which we also implement. If we implement those algorithms and didn't need those ctrls, I don't see why an external engine would need them. So, this is perhaps more likely for algorithms implemented by engines that we don't have internally. This seems to be much rarer - GOST is the only example I can think of.
So, while it is possible and I can't rule it out, I think it is quite unlikely.

Reply:
Heh, I just looked back at 1.0.2, where `engines/ccgost` still exists, and that version of it doesn't use MICALG at all...

Reply:
Interesting